keysign/gpgmeh.py

```diff
@@ -471,3 +474,34 @@ def sign_keydata_and_encrypt(keydata, error_cb=None, homedir=None):
                                                always_trust=True,
                                                sign=False)
                 yield (UID.from_gpgme(uid), ciphertext)
+
+
+def import_signature(encrypted_sig, homedir=None):
+    # ctx = gpg.Context()
+    ctx = DirectoryContext(homedir)
+    signature = ctx.decrypt(encrypted_sig)
+
+    # Try Seahorse DBus
+    name = "org.gnome.seahorse"
+    path = "/org/gnome/seahorse/keys"
+    bus = dbus.SessionBus()
+    result = []
+
+    try:
+        proxy = bus.get_object(name, path)
+    except dbus.exceptions.DBusException:
+        log.debug("Seahorse DBus is not available")
+    else:
+        iface = "org.gnome.seahorse.KeyService"
+        gpg_iface = dbus.Interface(proxy, iface)
+        payload = base64.b64encode(signature[0]).decode('latin-1')
+        payload = '\n'.join(payload[i:(i + 64)] for i in range(0, len(payload), 64))
+        payload = "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\n" + payload + "\n-----END PGP PUBLIC KEY BLOCK-----"
+        result = gpg_iface.ImportKeys("openpgp", payload)
+
+    # If Seahorse failed we try op_import
+    if len(result) < 1:
+        ctx.op_import(signature[0])
+        result = ctx.op_import_result()
+        if len(result.imports) < 1:
+            raise GPGMEError
```

Comment:
is this valid Python 3 code? I thought that you cannot raise classes but only objects.

Reply:
Yes it is allowed https://stackoverflow.com/questions/19768515/is-there-a-difference-between-raising-exception-class-and-exception-instance

`raise GPGMEError` and `raise GPGMEError()` provides the same result.